### PR TITLE
[Analyzer] Ensure plugin sources are added to the sources manager

### LINKED ISSUE #8540: Sources vendored by plugins not working
👋 Aloha!

* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

ℹ Please replace these two lines with what you did.  

Upgraded CocoaPods to `1.6.0` from `1.5.3` and run `pod install` on a project using a source vendored by a plug-in (via the [`:source_provider`](https://github.com/CocoaPods/CocoaPods/blob/b918bc0fbcc622401a4c5c8fe298cfd95df1fb7a/lib/cocoapods/installer.rb#L638)).

## What did you expect to happen?

ℹ Please replace these two lines with what you expected to happen. 

The installation to succeed.

## What happened instead?

ℹ Please replace these two lines with of what happened instead.  
 
`pod install` fails because CocoaPods is unable to find the source vendored by the plugin.

```
Pod::StandardError - [Bug] Failed to find known source with the URL "protocol://my-source"
/ruby/gems/2.3.0/gems/cocoapods-core-1.6.0/lib/cocoapods-core/source/manager.rb:38:in `aggregate_for_dependency'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:406:in `aggregate_for_dependency'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:392:in `create_set_from_sources'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:361:in `find_cached_set'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:335:in `specifications_for_dependency'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:142:in `search_for'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:251:in `block in sort_dependencies'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:244:in `each'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:244:in `sort_by'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:244:in `sort_dependencies'
/ruby/gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/delegates/specification_provider.rb:53:in `block in sort_dependencies'
/ruby/gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/delegates/specification_provider.rb:70:in `with_no_such_dependency_error_handling'
/ruby/gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/delegates/specification_provider.rb:52:in `sort_dependencies'
/ruby/gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:288:in `initial_state'
/ruby/gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:210:in `start_resolution'
/ruby/gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/resolution.rb:168:in `resolve'
/ruby/gems/2.3.0/gems/molinillo-0.6.6/lib/molinillo/resolver.rb:43:in `resolve'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/resolver.rb:91:in `resolve'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/installer/analyzer.rb:913:in `block in resolve_dependencies'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/user_interface.rb:64:in `section'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/installer/analyzer.rb:908:in `resolve_dependencies'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/installer/analyzer.rb:114:in `analyze'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/installer.rb:266:in `analyze'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/installer.rb:174:in `block in resolve_dependencies'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/user_interface.rb:64:in `section'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/installer.rb:173:in `resolve_dependencies'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/installer.rb:136:in `install!'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/command/install.rb:48:in `run'
/ruby/gems/2.3.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/lib/cocoapods/command.rb:52:in `run'
/ruby/gems/2.3.0/gems/cocoapods-1.6.0/bin/pod:55:in `<top (required)>'
project/runtime/ruby2.3.x/gem_bin/pod:22:in `load'
project/runtime/ruby2.3.x/gem_bin/pod:22:in `<main>'
```

After a cursory look it looks like the `plugin_sources` [are not being added to the Configs pluging manager](https://github.com/CocoaPods/CocoaPods/blob/master/lib/cocoapods/installer/analyzer.rb#L176).

### PR DESCRIPTION
**Pull request by amorde**
Closes #8540

@fabiopelosin could you try this out on your project to confirm that it resolves the issue you are experiencing?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,7 +38,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
-* None.  
+* Fix a bug that prevented dependencies in a plugin source from resolving
```
Comment (dnkoutso) [inline CHANGELOG.md]: this not on the right spot.

Comment (amorde) [reply CHANGELOG.md]: ah my bad, good catch

File: lib/cocoapods/installer/analyzer.rb [hunk 2]
```diff
@@ -170,6 +170,9 @@ def sources
           end
           unless plugin_sources.empty?
             result.insert(0, *plugin_sources)
+            plugin_sources.each do |source|
+              config.sources_manager.add_source(source)
```
Comment (segiddins) [inline lib/cocoapods/installer/analyzer.rb]: should this be added to the global sources manager? should we make our own sources manager in the analyzer?

Comment (amorde) [reply lib/cocoapods/installer/analyzer.rb]: updated this to use a separate sources manager rather than the global one

**Comment by amorde** [comment]
Rebased, pointed to master (this was originally intended to be released as a minor 1.6.x release),  and updated the analyzer and resolver to handle separate source managers

**Comment by amorde** [comment]
Yay green again

**Comment by amorde** [comment]
I was (and still am) very confused, not sure I am deserving of the blind trust 😂 until someone comes up with an easy to reproduce bug _or_ the original bug author tests out the change, I guess we'll go with the test cases here

**Pull request merged**